test/e2e/cloudprovider/cloudprovider.go

```diff
@@ -0,0 +1,339 @@
+	testTimeout    = 10 * time.Minute
+)
+
+var (
+	labels = map[string]string{"app": "test-cp"}
+)
+
+func RunCloudProviderTests(ctx context.Context, t *testing.T, client ctrlruntimeclient.Client, cloudProvider string) error {
+	t.Helper()
+	t.Log("Testing cloud provider support...")
+
+	if err := createPVCWithStorage(ctx, t, client, cloudProvider); err != nil {
+		return err
+	}
+	if err := exposePVC(ctx, t, client, cloudProvider); err != nil {
+		return err
+	}
+
+	return cleanUp(ctx, t, client)
+}
+
+func createPVCWithStorage(ctx context.Context, t *testing.T, client ctrlruntimeclient.Client, cloudProvider string) error {
+	t.Helper()
+
+	if !supportsStorage(cloudProvider) {
+		t.Logf("Skipping storage tests because cloud provider %q is not supported.", cloudProvider)
+
+		return nil
+	}
+
+	t.Log("Testing storage support...")
+
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	if err := client.Create(ctx, ns); err != nil {
+		return fmt.Errorf("failed to create namespace: %w", err)
+	}
+
+	t.Log("Creating a StatefulSet with PVC...")
+
+	// Creating a simple StatefulSet with 1 replica which writes to the PV. That way we know if storage can be provisioned and consumed
+	set := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      statefulsetName,
+			Namespace: ns.Name,
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "nginx",
+							Image: "nginx",
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "web",
+									ContainerPort: 80,
+								},
+							},
+						},
+						{
+							Name:  "busybox",
+							Image: "k8s.gcr.io/busybox",
+							Args: []string{
+								"/bin/sh",
+								"-c",
+								`echo "alive" > /data/healthy; sleep 3600`,
+							},
+							ReadinessProbe: &corev1.Probe{
+								InitialDelaySeconds: 0,
+								SuccessThreshold:    3,
+								PeriodSeconds:       5,
+								TimeoutSeconds:      1,
+								FailureThreshold:    1,
+								Handler: corev1.Handler{
+									Exec: &corev1.ExecAction{
+										Command: []string{
+											"cat",
+											"/data/healthy",
+										},
+									},
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "data",
+									MountPath: "/data",
+								},
+							},
+						},
+					},
+				},
+			},
+			VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "data",
+					},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{corev1.ResourceStorage: resource.MustParse("1Gi")},
```

Comment:
I remember hetzner had minimum volume requirement of 10 gigs... Did this changed?

Reply:
Looks like yes. The Hetzner presubmit is passing, and this code is also used in KKP, so I assume they changed it.